src/logger.ts

```diff
@@ -41,23 +41,29 @@ export const debug_info = {
 export async function saveInterfaceStat(): Promise<void> {
   console.log(apiPerfLogData)
   try {
-    // eslint-disable-next-line prefer-const
-    let { api_name, tfinal, timestamp, nodeUrl, success, reason, hash } = apiPerfLogData[0]
-    // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-    let placeholders = `NULL, '${api_name}', '${tfinal}','${timestamp}', '${nodeUrl}', '${success}', '${reason}', '${hash}'`
-    let sql = 'INSERT INTO interface_stats VALUES (' + placeholders + ')'
-    for (let i = 1; i < apiPerfLogData.length; i++) {
-      // eslint-disable-next-line prefer-const
-      let { api_name, tfinal, timestamp, nodeUrl, success, reason, hash } = apiPerfLogData[i] // eslint-disable-line security/detect-object-injection
-
-      // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-      placeholders = `NULL, '${api_name}', '${tfinal}','${timestamp}', '${nodeUrl}', '${success}', '${reason}', '${hash}'`
-      sql = sql + `, (${placeholders})`
-    }
+    const stmt = db.prepare(`
+      INSERT INTO interface_stats 
+      ("api_name", "tfinal", "timestamp", "nodeUrl", "success", "reason", "hash") 
+      VALUES (?, ?, ?, ?, ?, ?, ?)
+    `)
+
+    const insertMany = db.transaction((items) => {
```

Comment:
We should handle errors and finalize the prepared statrement.  

```suggestion
    const insertMany = db.transaction((items) => {
```
```suggestion
   const insertMany = db.transaction((items) => {
  try {
    for (const item of items) {
      stmt.run(
        item.api_name,
        item.tfinal,
        item.timestamp,
        item.nodeUrl,
        item.success ? 1 : 0,
        item.reason || null,
        item.hash || null
      );
    }
  } catch (error) {
    console.error('Error in transaction, rolling back:', error);
    throw error
  } finally {
    stmt.finalize()
  }
});

```

Reply:
no use of finalize in better-sqllite3.  ignore